src/simulation/misc/Nightmare.ts

```diff
@@ -0,0 +1,170 @@
+	'Prayer potion(3)': [[2, 10], 5],
+	'Saradomin brew(3)': [[1, 10], 5],
+	'Zamorak brew(3)': [[1, 8], 5],
+	'Sanfew serum(3)': [[1, 11], 5],
+
+	Coins: [[2717, 41_575], 2]
+};
+const nonUniqueItemRanges = resolveNameBank(data);
+
+const NonUniqueTable = new SimpleTable<number>();
+for (const [id, data] of Object.entries(nonUniqueItemRanges)) {
+	NonUniqueTable.add(parseInt(id), data[1]);
+}
+
+const GearTable = new SimpleTable<string>()
+	.add("Inquisitor's mace")
+	.add("Inquisitor's great helm", 2)
+	.add("Inquisitor's hauberk", 2)
+	.add("Inquisitor's plateskirt", 2)
+	.add('Nightmare staff', 3);
+
+const OrbTable = new SimpleTable<string>()
+	.add('Eldritch orb')
+	.add('Volatile orb')
+	.add('Harmonised orb');
+
+const mvpTertiary = new LootTable()
+	.tertiary(190, 'Clue scroll (elite)')
+	.tertiary(1900, 'Jar of dreams')
+	.tertiary(3800, 'Little nightmare');
+
+const nonMvpTertiary = new LootTable()
+	.tertiary(200, 'Clue scroll (elite)')
+	.tertiary(2000, 'Jar of dreams')
+	.tertiary(4000, 'Little nightmare');
+
+class NightmareClass {
+	hp = 2400;
+
+	public rollNonUniqueLoot(percentage: number, isMvp: boolean): [number, number] {
+		const { item } = NonUniqueTable.roll();
+
+		const [range] = nonUniqueItemRanges[item];
+		let quantity = range[0];
+		quantity += randInt(0, calcPercentOfNum(percentage, range[1]));
+		if (isMvp) quantity *= 1.1;
+
+		return [item, Math.floor(quantity)];
+	}
+
+	public kill(
+		options: O.Readonly<NightmareOptions>
+	): {
+		[key: string]: ItemBank;
+	} {
+		const mvp = options.team.sort((a, b) => b.damageDone - a.damageDone)[0];
+
+		const parsedTeam = options.team.map(teamMember => ({
+			...teamMember,
+			percentDamage: Math.floor(calcWhatPercent(teamMember.damageDone, this.hp)),
+			mvp: mvp === teamMember
+		}));
+
+		const lootResult: LootBank = {};
+
+		// Construct a weighted table, where the weighting is the percent of the total HP that the team member has damaged,
+		// for example, dealing 50% off of the nightmares health will give a weighting of 50, so they're 5x more likely
+		// to get the unique than the person who dealt 10%. However, in most realistic scenarios, the team members will have
+		// done very similar amounts of damage, varying by only a few percent.
+		const WeightedUniqueTable = new SimpleTable<string>();
+		for (const teamMember of parsedTeam) {
+			lootResult[teamMember.id] = new Loot();
+
+			WeightedUniqueTable.add(teamMember.id, teamMember.percentDamage);
+		}
+
+		let gotInitialUnique = false;
+		function giveWeightedDrop(item: string): void {
+			const { item: recipient } = WeightedUniqueTable.roll();
+			lootResult[recipient].add(item);
+			if (!gotInitialUnique) gotInitialUnique = true;
+		}
+
+		if (roll(120)) {
+			giveWeightedDrop(GearTable.roll().item);
+		}
+
+		if (roll(600)) {
+			giveWeightedDrop(OrbTable.roll().item);
+		}
+
+		const secondRollChance = Math.min(75, parsedTeam.length - 5);
+		if (gotInitialUnique && secondRollChance > 0) {
```

Comment:
```suggestion
		if (secondRollChance > 0) {
```
The second roll happens no matter if a unique was awarded to a person or not. In reality one could get 4 unique drops however the chance of this is 1.436666e-14, which is just a so small number that it most likely never happen and never will happen.